hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpTopicsSubcommand.java

```diff
@@ -0,0 +1,200 @@
+
+    static void doit(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path topicsPath,
+            @NonNull final EmitSummary emitSummary,
+            @NonNull final Verbosity verbosity) {
+        new DumpTopicsSubcommand(state, topicsPath, emitSummary, verbosity).doit();
+    }
+
+    @NonNull
+    final SignedStateHolder state;
+
+    @NonNull
+    final Path topicsPath;
+
+    @NonNull
+    final EmitSummary emitSummary;
+
+    @NonNull
+    final Verbosity verbosity;
+
+    DumpTopicsSubcommand(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path topicsPath,
+            @NonNull final EmitSummary emitSummary,
+            @NonNull final Verbosity verbosity) {
+        requireNonNull(state, "state");
+        requireNonNull(topicsPath, "topicsPath");
+        requireNonNull(emitSummary, "emitSummary");
+        requireNonNull(verbosity, "verbosity");
+
+        this.state = state;
+        this.topicsPath = topicsPath;
+        this.emitSummary = emitSummary;
+        this.verbosity = verbosity;
+    }
+
+    void doit() {
+        final var topicsStore = state.getTopics();
+        System.out.printf("=== %d topics ===%n", topicsStore.size());
+
+        final var allTopics = gatherTopics(topicsStore);
+
+        int reportSize;
+        try (@NonNull final var writer = new Writer(topicsPath)) {
+            if (emitSummary == EmitSummary.YES) reportSummary(writer, allTopics);
+            reportOnTopics(writer, allTopics);
+            reportSize = writer.getSize();
+        }
+
+        System.out.printf("=== topics report is %d bytes%n", reportSize);
+    }
+
+    @SuppressWarnings(
+            "java:S6218") // "Equals/hashcode method should be overridden in records containing array fields" - this
+    record Topic(
+            int number,
+            @NonNull String memo,
+            @NonNull RichInstant expirationTimestamp,
+            boolean deleted,
+            @NonNull JKey adminKey,
+            @NonNull JKey submitKey,
+            @NonNull byte[] runningHash,
+            long sequenceNumber,
+            long autoRenewDurationSeconds,
+            @Nullable EntityId autoRenewAccountId) {
+        Topic(@NonNull final MerkleTopic topic) {
+            this(
+                    topic.getNumber(),
+                    topic.getMemo(),
+                    topic.getExpirationTimestamp(),
+                    topic.isDeleted(),
+                    topic.getAdminKey(),
+                    topic.getSubmitKey(),
+                    null != topic.getRunningHash() ? topic.getRunningHash() : EMPTY_BYTES,
+                    topic.getSequenceNumber(),
+                    topic.getAutoRenewDurationSeconds(),
+                    topic.getAutoRenewAccountId());
+            Objects.requireNonNull(memo, "memo");
+            Objects.requireNonNull(adminKey, "adminKey");
+            Objects.requireNonNull(submitKey, "submitKey");
```

Comment:
is the submit key really require non null?  I thought they were optional for the user to supply.  (Though that's independent of whether the _representation_ is null.)

also is memo really require non null?  other memo fields in other semantic entities are optional and can be null.  (IIRC)

(for that matter I'm not even sure admin key is require non null! is that enforced by services when you create/udpate the key? wish I knew ...)

Reply:
referring to [hedera docs(create a topic)](https://docs.hedera.com/hedera/sdks-and-apis/sdks/consensus-service/create-a-topic) looks like neither of them is required, so assume we have to remove all of the `requireNonNull`s?
![image](https://github.com/hashgraph/hedera-services/assets/77534278/11b52b2d-b49f-493a-8803-79aaf71b7a63)

Reply:
Remember to use `getNullableFormatter` down at the definition of `FieldFormatters` too...

Reply:
Actually, now looking at `MerkleTopic` again, I see the getters for admin key and submit key will _not_ return `null` but instead will always return something - an empty key list in the case no key was supplied.  (Sorry, should have looked earlier when I originally made the comment.)

(This is different from, say, `MerkleToken`, which returns `Optional<JKey>` (which IMO is a better API, but whatever).)

So you can leave the `requireNonNull` there.

(This may change when the modular representation is added.  I haven't looked but it's possible the modular representation may return null, or may. return `Optional`.  We'll have to see ...)

Reply:
Left the requireNonNulls and fixed by using `getNullableFormatter` for `adminKey` and `supplyKey`